src/nerpa_pipeline/handle_rban.py

```diff
@@ -105,41 +111,102 @@ def parse_as_simple_cycle(G: nx.DiGraph) -> Union[List[int], None]:
         return None
 
 
-def putative_backbones(G: nx.DiGraph, min_nodes: int=2) -> Tuple[List[List[int]],
-                                                                 List[List[int]]]:
-    '''
-    Tries to parse each componenet of G either as a simple cycle or as a hamiltonian path
-    '''
+class BackboneSequence(NamedTuple):
+    type: Literal['PATH', 'CYCLE']
+    node_idxs: List[int]
+
+
+def putative_backbones(G: nx.DiGraph, min_nodes: int=2) -> List[BackboneSequence]:
+    ''' Tries to parse each component of G either as a simple cycle or as a hamiltonian path '''
     breakage_points = [node for node in G.nodes()
                        if G.in_degree(node) == 0 and G.out_degree(node) > 1 \
                        or G.in_degree(node) > 1 and G.out_degree(node) == 0]  # sinks and sources of degree > 1
     G.remove_nodes_from(breakage_points)  # modifying an argument makes me anxious but the next TODO should help it
     # TODO: perform removing high degree sinks and sources at the same time as removing other nodes and bonds
-    
-    cycles, paths = [], []
+
+    backbone_sequences = []
     for component in nx.connected_components(nx.Graph(G)):
         if len(component) < min_nodes:
             continue
 
         Gs = G.subgraph(component)
         if cycle := parse_as_simple_cycle(Gs):
-            cycles.append(cycle)
+            backbone_sequences.append(BackboneSequence(type='CYCLE',
+                                                       node_idxs=cycle))
         else:
-            paths += [ham_path for u in Gs
-                      if (ham_path := hamiltonian_path(Gs, u))]
+            backbone_sequences.extend(BackboneSequence(type='PATH',
+                                                       node_idxs=ham_path)
+                                      for u in Gs
+                                      if (ham_path := hamiltonian_path(Gs, u)))
```

Comment:
Potentially can lead to several paths per component (eg. when rban fails to detach two monomers)